src/pages/Challenge Management/challengeOverview/ChallengeOverview.js

```diff
@@ -1,42 +1,102 @@
-import React from "react";
-import { StyleSheet, SafeAreaView, ScrollView, Text } from "react-native";
+import React, { useEffect, useState } from "react";
+import { View, SafeAreaView, ScrollView, Text } from "react-native";
 import PopupContainer from "../../../components/Contaner/PopupContainer";
 import BigHeaderBackground from "../../../components/HeaderBackground/HeaderBackground";
 import Overview from "../../../components/Challenges/ChallengeOverview/Overview";
 import DateList from "../../../components/Challenges/DateList/DateList";
 import Hr from "../../../components/HorizontalLine/Hr";
 import styles from "./styles";
 import { ChallengeConstants } from "../../../util/Constants/ChallengeConstants";
+import { getChallenge } from "../../../api/challenge.api";
+import { useNavigation } from "@react-navigation/native";
+import { CommonConstants } from "../../../util/Constants/CommonConstants";
 
 export default function ChallengeOverview({route}) {
-  console.log(route)
+  const challengeId = route.params.id
+  const [challenge, setChallenge] = useState({});
+  const navigation = useNavigation();
+
+  useEffect(() => {
+    fetchChallenge()
+  }, []);
+
+  function handleOnClick() {
+    navigation.navigate(CommonConstants.CHALLENGE_PROGRESS_PATH);
+  }
+
+  // function getProgressProps() 
+  //   const props;
+  //   if (!challenge.isStarted || challenge.startDate) {
+  //     props.daysLeft = challenge.duration;
+  //     props.tasks = challenge.task[0];
+  //   } else {
+  //     const new Date(challenge.startDate.split('T')[0]) - new Date(challenge.tasks[0].date.split('T')[0]))/ (1000 * 60 * 60 * 24)
+  //   }
+  // }
```

Comment:
Check this maybe a mistake?

Reply:
not a mistake..  I started working on it but did not finish. so commented out so that i wont forget this